src/ui/Icon/Icon.ts

```diff
@@ -93,6 +101,24 @@ export class Icon extends Component {
     this.result = this.result || this.resolveResult();
     Assert.exists(this.result);
 
+    if (this.options.condition != null) {
+      var conditionFunction = new Function('obj', 'with(obj||{}){return ' + this.options.condition + '}');
+      if (conditionFunction(this.result)) {
+        this.initialize(element, bindings, false);
+      } else {
+        this.initialize(element, bindings, true);
+      }
+    } else {
+      this.initialize(element, bindings, false);
+    }
+  }
+
+  private initialize(element: HTMLElement, bindings: IComponentBindings, removeFromParent: boolean) {
+    // Completely remove the element to ease stuff such as adding separators in CSS
+    if (this.element.parentElement != null && removeFromParent) {
```

Comment:
I would not group this logic into the `initialize` method, but rather create a method to use when you want this behaviour. e.g.

```
private removeElementAndInitialize(element: HTMLElement, bindings: IComponentBindings) {
    if (...)
    this.initialize(element, bindings)
}
```

This makes the code more readable compared to calling `initialize` with a cryptic `true/false` third parameter.

Reply:
Agreed